lib/components/javascript.js

```diff
@@ -0,0 +1,32 @@
+/* eslint-disable no-undef */
+// this function will be stringified as part of the generated types and is not supposed to be used internally
+const proxyAccessFunction = function (namespace, name, opts = {}) {
+    const { target, customProps } = { target: {}, customProps: [], ...opts }
+    const fq = namespace !== '' ? `${namespace}.${name}` : name
+    let _entity
+    return new Proxy(target, {
+        get: function (target, prop) {
+            // we already know the name so we skip the cds.entities proxy access
+            if (prop === 'name') return fq
+            // custom properties access on 'target' as well as cached _entity property access goes here
+            if (Object.prototype.hasOwnProperty.call(target, prop)) return target[prop]
+            // inline enums have to be caught here for first time access, as they do not exist on the entity
+            if (customProps.length && customProps.includes(prop)) return target[prop]
+            // last but not least we pass the property access to cds.entities
+            if (cds.entities) {
+                const entity = _entity ? _entity : (_entity = cds.entities(namespace)[name])
```

Comment:
```suggestion
                entity ??= cds.entities(namespace)[name]
```
I guess that should result in the same, no?

Reply:
Yes